test/es-module/test-esm-detect-ambiguous.mjs

```diff
@@ -282,7 +291,10 @@ describe('Module syntax detection', { concurrency: !process.env.TEST_PARALLEL },
         'function fn() { await Promise.resolve(); } fn();',
       ]);
 
-      match(stderr, /SyntaxError: await is only valid in async function/);
+      match(
+        stderr,
+        /Top-level await is not supported in CommonJS modules/
```

Comment:
This test is titled “still throws on `await` in an ordinary sync function”. Based on the change you made to the assertion, your PR breaks this test.